internal/server/worker_auth.go

```diff
@@ -50,6 +51,21 @@ func (w *WorkerAuth) decrypt(ctx context.Context, cipher wrapping.Wrapper) error
 	return nil
 }
 
+func (w *WorkerAuth) compare(other *WorkerAuth) bool {
+	switch {
+	case w.WorkerId != other.WorkerId:
+		return false
+	case !bytes.EqualFold(w.WorkerEncryptionPubKey, other.WorkerEncryptionPubKey):
```

Comment:
Is using `EqualFold` actually correct here? I don't think we want to be doing any normalization as it could affect the underlying data, shouldn't we just use `Equal`?

Reply:
Ah thanks! GoLand autocomplete bit me on this one